DriftCorrection class and supporting infrastructure

cophus opened the pull request:
Added DriftCorrection, Dataset2d, Dataset3d, some utils, tweaks to some core functions

gvarnavi reviewed (approved):
Great job!

Did a fair bit of cleanup, but the functionality is mostly unchanged:

- I renamed `_initialize()` to `preprocess()` and call it explicitly after init
- I implemented chaining
- I added `show_images` and `show_knots` to all methods to plot intermediate results

It now works like so:

```python
drift = em.imaging.DriftCorrection.from_data(
    images = [
        im0, 
        im1,
    ],
    scan_direction_degrees = [0, 90],
    number_knots = 1,
    pad_fraction = 0.25,
).preprocess(
    show_images=True,
).align_affine(
    step = 0.02,
    num_tests = 9,
    show_images=True,
).align_nonrigid(
    regularization_sigma_px = 0.5,
    max_optimize_iterations = 10,
    regularization_max_shift_px = 1.0,
    solve_individual_rows=True,
    show_images=True,
)
```

Attaching updated notebook: [drift11_gv.ipynb.zip](https://github.com/user-attachments/files/20235772/drift11_gv.ipynb.zip)

cophus commented:
I tweaked a lot of the `preprocess()`. and `show_images` settings / defaults, added `show_merged` as the default plotting outputs. Also fixed a few bugs like Fourier downsampling changing the intensity, and I added grids of titles.

cophus commented:
A minimal example showing the initial processing options moved to `.preprocess()`

```
drift = em.imaging.DriftCorrection.from_data(
    images = [
        im0, 
        im1,
    ],
    scan_direction_degrees = [0, 90],
).preprocess(
    pad_fraction = 0.25,
    pad_value = 'median',
    kde_sigma = 0.5,
    show_merged = True,
).align_affine(
    step = 0.02,
    num_tests = 11,
).align_nonrigid(
    regularization_sigma_px = 0.5,
    regularization_max_shift_px = 2.0,
)
```
